Documentation/CRDs/Cluster/external-cluster.md

```diff
@@ -54,51 +56,60 @@ In order to configure an external Ceph cluster with Rook, we need to extract som
+To enable multi-tenancy, run the script with the `--restricted-auth-permission` flag and pass the mandatory flags with it,
+It will generate the secrets which you can use for creating new `Consumer cluster` deployment using the same `Source cluster`(ceph cluster).
+So you would be running different isolated consumer clusters on top of single `Source cluster`.
 
-- `--upgrade`: (optional) Upgrades the 'Ceph CSI keyrings (For example: client.csi-cephfs-provisioner) with new permissions needed for the new cluster version and older permission will still be applied.
+!!! note
+    Restricting the csi-users per pool, and per cluster will require creating new csi-users and new secrets for that csi-users.
+    So apply these secrets only to new `Consumer cluster` deployment while using the same `Source cluster`.
 
-    Sample run:
-    ```console
-    python3 create-external-cluster-resources.py --upgrade
-    ```
-    This will upgrade all the default csi-users(non-restricted)
+#### Sample run
 
-    Restricted users created using --restricted-auth-permission flag (For example: client.csi-rbd-node-rookStorage-replicapool) need to pass mandatory flags: '--rbd-data-pool-name(if it is a rbd user), --cluster-name and --run-as-user' flags while upgrading, in case of cephfs users if you have passed --cephfs-filesystem-name flag while creating csi-users then while upgrading it will be mandatory too
+```console
+python3 create-external-cluster-resources.py --cephfs-filesystem-name <filesystem-name> --rbd-data-pool-name <pool_name> --cluster-name <cluster-name> --restricted-auth-permission true --format <bash> --rgw-endpoint <rgw_endpoin> --namespace <rook-ceph-external>
+```
 
-    Sample run:
-    ```console
-    python3 create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --cluster-name rookstorage --run-as-user client.csi-rbd-node-rookStorage-replicapool
-    ```
+#### Upgrade Sample Run
 
-!!! note
-    An existing non-restricted user cannot be downgraded to a restricted user by upgrading. Admin needs to create a new restricted user for this by re-running the script.The upgrade flag should only be used to append new permissions to users. It shouldn't be used for changing a csi user already applied permissions. For example, you shouldn't change the pool(s) a user has access to.
+1) If consumer cluster doesn't have restricted caps,This will upgrade all the default csi-users(non-restricted):
+```console
+python3 create-external-cluster-resources.py --upgrade
+```
 
-2. Copy the bash output.
+2) If the consumer cluster has restricted caps:
+Restricted users created using --restricted-auth-permission flag (For example: client.csi-rbd-node-rookStorage-replicapool) need to pass mandatory flags: '--rbd-data-pool-name(if it is a rbd user), --cluster-name and --run-as-user' flags while upgrading, in case of cephfs users if you have passed --cephfs-filesystem-name flag while creating csi-users then while upgrading it will be mandatory too
```

Comment:
I don't understand this example, is there a typo in "(For example: client.csi-rbd-node-rookStorage-replicapool)"?
```suggestion
Restricted users created using `--restricted-auth-permission` flag (For example: client.csi-rbd-node-rookStorage-replicapool) need to pass mandatory flags: '--rbd-data-pool-name` (if it is a rbd user), `--cluster-name` and `--run-as-user' flags while upgrading, in case of cephfs users if you have passed `--cephfs-filesystem-name` flag while creating csi-users then while upgrading it will be mandatory too
```